compiler+runtime/include/cpp/jank/runtime/core/make_box.hpp

```diff
@@ -82,6 +82,14 @@ namespace jank::runtime
     return make_box<runtime::obj::persistent_list>(l);
   }
 
+  template <typename T>
+  requires std::is_pointer_v<T>
+  [[gnu::always_inline, gnu::flatten, gnu::hot]]
+  inline auto make_box(T const d)
+  {
+    return make_box<obj::native_pointer_wrapper>(static_cast<void *>(d));
+  }
```

Comment:
I thought about this, and it seemed clever at first, but I think it's too clever. Anything we don't catch in another overload gets here, even if we were previously relying on implicit conversions. Let's remove this and rely on explicit `make_box<obj::native_pointer_wrapper>()` for now.